design-documents/csp.md

```diff
@@ -119,17 +113,87 @@ It is possible to combine nonce and domain-list based whitelisting for CSP.
  
 Magento will have to generate unique nonce (using _Magento\Framework\Math\Random_) for each response and developers will
 be able to use a presentation layer class to render nonce to their explicit inline scripts
-(similar to how form key is being added to templates). WIth _nonce_ enabled
+(similar to how form key is being added to templates). With _nonce_ enabled
 _Magento\Framework\View\Page\Config\Renderer_ will add nonce to all css/js assets being added to a page. With _nonce_
 disabled the presentation class used inside templates will not add nonce when called and _Renderer_ will not do it as well.
  
-To make it easier for developers and don't force them to include the nonce-helper to all of their blocks it will
-be available inside _.phtml_ templates via local `$nonce` variable.
+Additional effort will have to go to ensure proper nonce used/regenrated when page cache is enabled. It is possible to
```

Comment:
Maybe it would be good to have a more concrete understanding that this can actually be done. As stated by @DrewML  Varnish support is integral to the success of this feature.

Reply:
@nathanjosiah this helper will be used for hashes/dynamic origin whitelisting depending on the CSP mode selected by merchant so we'd have to have it. There is no conflict between hashes/whitelisted origins and using varnish

Reply:
Usually, Varnish stores the gzipped content in the cache. It means that Varnish will be forced to decompress every cache entry, modify it and compress again for each HTTP request. This will definitely consume more system resources.

Also, we need to support third-party caching providers like Fastly.

Thus, please expand this section with a strategy for third-party providers and another impact on env, setup, etc.

Reply:
using nonce alongside page cache doesn't seem viable right now, updated the section